src/foam/nanos/auth/AuthenticatedUserDAO.java

```diff
@@ -64,17 +63,36 @@ public FObject put_(X x, FObject obj) {
     if ( ! SafetyUtil.equals(toPut.getId(), user.getId()) &&
       ! auth.check(x, GLOBAL_USER_UPDATE) &&
       ! auth.check(x, GLOBAL_SPID_UPDATE) &&
-      ! auth.check(x, "spid.update." + user.getSpid()) ) {
-      throw new RuntimeException("Unable to update user");
+      ! auth.check(x, "spid.update." + user.getSpid())
+    ) {
+      throw new AuthorizationException("You do not have permission to update this user.");
     }
 
     // set spid if not set
-    if ( SafetyUtil.isEmpty(toPut.getSpid()) &&
-        ! SafetyUtil.isEmpty(user.getSpid()) ) {
-      toPut.setSpid(user.getSpid());
+    if ( SafetyUtil.isEmpty((String) newUser.getSpid()) &&
+        ! SafetyUtil.isEmpty((String) user.getSpid()) ) {
+      newUser.setSpid(user.getSpid());
     }
 
-    return super.put_(x, toPut);
+    User oldUser = (User) super.find_(x, newUser.getId());
```

Comment:
you can modify the `isExistingUser` function to return you the user if it exists. You should probably change the name of the function to something like `getExistingUser`